### ISSUE TITLE
2.17 - outputPaths vendor JS option triggers "Could not find module `ember-resolver`" error in browser

### ISSUE THREAD

**Issue by GCheung55** [ember-cli/ember-cli#7518]
With 2.17+, `ember-resolver` is not imported when `outputPaths` contains a vendor JS config option in `ember-cli-build.js`

```javascript
let app = new EmberApp(defaults, {
  // Add options here
  outputPaths: {
    // app: {
    //    js: '/assets/scripts/my-app.js'
    // },
    vendor: {
      js: '/assets/scripts/vendor.js'
    }
  }
});
```

I have a demo of this happening in this repo: https://github.com/GCheung55/ember-upgrade-repo

---

Output from `ember version --verbose && npm --version && yarn --version`:
```
ember-cli: 2.17.1
http_parser: 2.7.0
node: 6.11.4
v8: 5.1.281.108
uv: 1.11.0
zlib: 1.2.11
ares: 1.10.1-DEV
icu: 58.2
modules: 48
openssl: 1.0.2l
os: darwin x64
3.10.10
1.3.2
```

**Comment by rwjblue** [comment]
Thank you for reporting!

**Comment by kevinansfield** [comment]
I ran into this too. Interestingly development builds were fine so it went undetected for quite a while, in our case it was only the production build that was throwing the `ember-resolver` error

**Comment by SergeAstapov** [comment]
Ran into the same issue while upgrading Ember CLI from 2.16 to 2.18.
And we have customized outputPath for `vendor.js` file via `outputPaths` config option due to some restrictions in infrastructure.

Applying change from #7519 makes Ember CLI happy.

Is it possible to release as a patch for 2.18?

**Comment by twokul** [comment]
https://github.com/ember-cli/ember-cli/pull/7610

**Comment by twokul** [comment]
I believe this could be closed as [v3.0.0](https://github.com/ember-cli/ember-cli/releases/tag/v3.0.0) was released with the fix included.

**Comment by gnapse** [comment]
But there should be a fix in the 2.x version range, don't you think?

**Comment by rohmann** [comment]
As @gnapse has pointed out this is still broken in 2.18.2. I'm working in a hybrid setup that runs an Ember Application inside of another project where we happen to have gulp available. This hack using [`gulp-replace`](https://www.npmjs.com/package/gulp-replace) helped me get things working again.

```
gulp.task('patch-ember', function() {
  return gulp.src(['node_modules/ember-cli/lib/broccoli/ember-app.js'])
    .pipe(replace('DEFAULT_CONFIG.outputPaths.vendor.js', function() {
      return "vendorFilePath";
    }))
    .pipe(gulp.dest('node_modules/ember-cli/lib/broccoli/'));
});
```

**Comment by avdv** [comment]
> I believe this could be closed as [v3.0.0](https://github.com/ember-cli/ember-cli/releases/tag/v3.0.0) was released with the fix included.

@twokul do you mean that we should use ember-cli 3.0.0 with ember source / library at 2.18.x?

**Comment by givanse** [comment]
@avdv I tried that combo and it didn't.
